internal/postgres/process.go

```diff
@@ -127,10 +132,7 @@ func readAndParseChunk(conv *Conv, r *internal.Reader) ([]byte, []nodes.Node) {
 			conv.stats.reparsed++
 		}
 		if r.EOF {
-			if len(l) != 0 {
-				fmt.Printf("Error parsing last %d line(s) of input\n", len(l))
-			}
-			return nil, nil
+			return nil, nil, fmt.Errorf("Error parsing input: %q", l)
```

Comment:
'l' could be huge: maybe MB of data.
Probably don't want to put it all in the error message.
Either don't put it in, or limit it.

Reply:
Thanks for pointing out. I changed it back to use `len(l)` for the error info. 

One question I have with this method is that e.g., use the following example: 

```sql
this is not a parsable file;

CREATE TABLE public.testusers (
    id integer NOT NULL,
    active boolean,
    name text
);

ALTER TABLE public.testusers OWNER TO postgres;

COPY public.testusers (id, active, name) FROM stdin;
1	t	foo
2	f	bar
\.

ALTER TABLE ONLY public.testusers
    ADD CONSTRAINT testusers_pkey PRIMARY KEY (id);
```

The first line is not a valid statement, but in this case, `l` will still try to include more statements and keep passing it to `pg_query.Parse(string(s))` until containing the content of the entire file. If the entire file is in GB size, we got an error at first place and can we fail it quickly? 

I understand that there might be cases the statement itself contains `;`. I just wonder if there is a better solution for this.

Reply:
Yes, that's correct: if we hit an error, we'll keep re-trying.
And, as you say, if the error is on the first line of the file, we'll eventually try to parse the entire file. If its big, that won't end well -- we can probably only parse files up to around a few hundred MB because of the overheads of the pg AST.

There are a couple of options:
a) we could put a limit on the amount of the file we try to parse: say bail out if we're trying to parse more than a few MB.
b) try to do something more intelligent for recognizing end-of-statement: this is hard, and potentially very fragile.
c) don't use pg_dump output and instead use more direct introspection into the PostgreSQL DB.

Longer-term, I think option (c) makes the most sense. But happy to do a quick version of (a) in the short-term. I don't think (b) is a great option.